File: solutions/lua/house/1/house.lua
local verses = {
  { that = "lay in",      the = "house that Jack built." },
  { that = "ate",         the = "malt" },
  { that = "killed",      the = "rat" },
  { that = "worried",     the = "cat" },
  { that = "tossed",      the = "dog" },
  { that = "milked",      the = "cow with the crumpled horn" },
  { that = "kissed",      the = "maiden all forlorn" },
  { that = "married",     the = "man all tattered and torn" },
  { that = "woke",        the = "priest all shaven and shorn" },
  { that = "kept",        the = "rooster that crowed in the morn" },
  { that = "belonged to", the = "farmer sowing his corn" },
  { that = "",            the = "horse and the hound and the horn" },
}

local house = {}
house.verse = function(which)
  local result = ("This is the %s"):format(verses[which].the)
  for i = which - 1, 1, -1 do
    result = result .. ("\nthat %s the %s"):format(verses[i].that, verses[i].the)
  end
  return result
end

house.recite = function()
  local result = setmetatable({}, { __index = table })
  for i in ipairs(verses) do result:insert(house.verse(i)) end
  return result:concat("\n")
end

return house

File: solutions/swift/house/1/Sources/House/House.swift
let verses: [(String, String)] = [
  ("", "horse and the hound and the horn"),
  ("belonged to", "farmer sowing his corn"),
  ("kept", "rooster that crowed in the morn"),
  ("woke", "priest all shaven and shorn"),
  ("married", "man all tattered and torn"),
  ("kissed", "maiden all forlorn"),
  ("milked", "cow with the crumpled horn"),
  ("tossed", "dog"),
  ("worried", "cat"),
  ("killed", "rat"),
  ("ate", "malt"),
  ("lay in", "house that Jack built."),
]

class House {
  static func recite(start: Int, end: Int) -> String {
    var result = ""
    for i in start - 1..<end {
      result += "This is the \(verses[verses.count - i - 1].1)"
      for verse in verses[verses.count - i..<verses.count] {
        result += " that \(verse.0) the \(verse.1)"
      }
      if i < end - 1 { result += "\n" }
    }
    return result
  }
}
